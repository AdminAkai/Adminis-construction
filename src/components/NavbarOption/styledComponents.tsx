import styled from 'styled-components'

export const NavbarOptionContainer = styled.div`
  margin-left: 16px;
`
