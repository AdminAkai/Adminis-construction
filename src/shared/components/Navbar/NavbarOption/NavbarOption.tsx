import { FC } from 'react'

import CustomLink from 'src/shared/components/CustomLink'

type NavbarOptionProps = {
  label: string
  route: string
  ariaLabel: string
}

const NavbarOption: FC<NavbarOptionProps> = ({ label, route, ariaLabel }) => {
  return (
    <CustomLink to={route} ariaLabel={ariaLabel} nav>
      {label}
    </CustomLink>
  )
}

export default NavbarOption
