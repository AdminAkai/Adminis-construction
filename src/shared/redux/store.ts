import { configureStore } from '@reduxjs/toolkit'
import { combineReducers } from 'redux'
import { useDispatch, TypedUseSelectorHook, useSelector } from 'react-redux'
import createSagaMiddleware from '@redux-saga/core'
import { all } from 'redux-saga/effects'
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist'
import storage from 'redux-persist/lib/storage'

import landingSagas from 'src/features/Landing/redux/landingSagas'
import landingSlice from 'src/features/Landing/redux/landingSlice'

import settingsSlice from './settingsSlice'

const sagas = createSagaMiddleware()

const persistConfig = {
  key: 'root',
  storage,
}

const reducers = combineReducers({
  settings: settingsSlice,
  landing: landingSlice,
})

const persistedReducer = persistReducer(persistConfig, reducers)

const middleWares = [sagas]

const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) => [
    ...getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
    ...middleWares,
  ],
})

export type RootState = ReturnType<typeof store.getState>

export type AppDispatch = typeof store.dispatch
type DispatchFunc = () => AppDispatch
export const useAppDispatch: DispatchFunc = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector

function* rootSaga() {
  yield all([...landingSagas])
}

sagas.run(rootSaga)

export const persistor = persistStore(store)

export default store
