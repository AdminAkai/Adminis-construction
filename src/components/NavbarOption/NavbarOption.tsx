import { FC, memo } from 'react'
import { useLocation } from 'react-router-dom'

import CustomLink from 'src/components/CustomLink'

import { NavbarOptionContainer } from './styledComponents'

interface INavbarOptionProps {
  label: string
  route: string
}

const NavbarOption: FC<INavbarOptionProps> = ({ label, route }) => {
  const { pathname } = useLocation()

  return (
    <NavbarOptionContainer>
      <CustomLink $isActive={pathname === route} to={route} $nav>
        {label}
      </CustomLink>
    </NavbarOptionContainer>
  )
}

const MemoizedOption = memo(NavbarOption)

export default MemoizedOption
