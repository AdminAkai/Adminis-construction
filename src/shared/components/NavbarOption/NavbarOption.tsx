import { FC, memo } from 'react'
import { useLocation } from 'react-router-dom'

import CustomLink from 'src/shared/components/CustomLink'

import { NavbarOptionContainer } from './styledComponents'

interface INavbarOptionProps {
  label: string
  route: string
  ariaLabel: string
}

const NavbarOption: FC<INavbarOptionProps> = ({ label, route, ariaLabel }) => {
  const { pathname } = useLocation()

  return (
    <NavbarOptionContainer>
      <CustomLink
        $isActive={pathname === route}
        to={route}
        $nav
        ariaLabel={ariaLabel}
      >
        {label}
      </CustomLink>
    </NavbarOptionContainer>
  )
}

const MemoizedOption = memo(NavbarOption)

export default MemoizedOption
