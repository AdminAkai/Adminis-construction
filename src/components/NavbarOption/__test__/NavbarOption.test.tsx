import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import '@testing-library/jest-dom'

import store from 'src/redux/store.ts'

import NavbarOption from '../NavbarOption.tsx'

const label: string = 'About'
const route: string = '/about'

describe('Navbar Option tests', () => {
  it('Make sure Navbar Option renders label and route', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <NavbarOption label={label} route={route} />
        </BrowserRouter>
      </Provider>
    )

    const navOption = screen.getByRole('link', { name: 'About' })
    expect(navOption).toBeInTheDocument()
    expect(navOption).toHaveTextContent('About')
    expect(navOption).toHaveAttribute('href', '/about')
  })

  it('Navbar Option snapshot renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <NavbarOption label={label} route={route} />
          </BrowserRouter>
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
