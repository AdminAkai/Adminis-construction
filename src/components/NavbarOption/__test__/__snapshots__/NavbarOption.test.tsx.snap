// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar tests Navbar renders correctly 1`] = `
<div
  className="sc-guDMob bTtonf"
>
  <a
    className="sc-beyTiQ igHIIw"
    href="/about"
    onClick={[Function]}
  >
    About
  </a>
</div>
`;
